### PR TITLE
[ADD] Debug Function with Variadic Functions

### PR DESCRIPTION
**Pull request by rajivnayanc**
This commit adds two variadic functions based debugger, which helps in better readability of the code. One functions end with a newline, and one doesn't. In the functions, we can pass any number of arguments to the functions that will print them space-separated when DEBUG Flag is defined. Also, build.sh is added for easily move library to samples folder.

### CODE REVIEW

File: build.sh
Comment (krshrimali) [inline build.sh:5]: Do you think having `cmake .. && make` makes more sense? If any one of them fails, then the other should ideally not run as well.

File: samples/FileManager.hpp
Comment (krshrimali) [inline samples/FileManager.hpp:75]: Nice!

File: src/FileManager.cpp
Comment (krshrimali) [inline src/FileManager.cpp:35]: Thanks for being thorough.

File: src/FileManager.cpp
Comment (krshrimali) [inline src/FileManager.cpp:42]: nit: should this be: `"dirs in: "` instead of `"dirs in:` (note the extra space at the end)? Or does the variadic template adds spaces at the end every time?

File: src/FileManager.cpp
Comment (krshrimali) [inline src/FileManager.cpp:179]: Are we missing `ifdef` and `#endif` here?

File: src/FileManager.cpp
Comment (krshrimali) [inline src/FileManager.cpp:191]: nit: let's remove space just before semicolon.

File: src/FileManager.cpp
Comment (krshrimali) [inline src/FileManager.cpp:198]: nit: let's add spaces here: `std::cout << "\n";` instead of `std::cout<<"\n";`.

File: src/FileManager.cpp
Comment (krshrimali) [inline src/FileManager.cpp:203]: same nit as above: let's remove space before semicolon here.

**Review by krshrimali — APPROVED** [review]
Thanks, @rajivnayanc - good additions. Looks neat, and thank you for changing all the macros to using variadic templates. I have suggested a few changes, which are mostly nits. Approving the PR but will appreciate if those nits can be addressed before we merge this.

Thank you for your contribution. 👏🏻 🎉

File: build.sh
Comment (rajivnayanc) [reply build.sh:5]: Yes. I have updated this in next commit.

File: src/FileManager.cpp
Comment (rajivnayanc) [reply src/FileManager.cpp:42]: The variadic template adds a space between two variables every time.

File: src/FileManager.cpp
Comment (rajivnayanc) [reply src/FileManager.cpp:179]: No, `#ifdef` and `#endif` are being added in the variadic template itself so that the code looks cleaner here.

**Comment by rajivnayanc** [comment]
I have updated the code wherever it was applicable. Please have a look. Thanks for the suggestions. 

cc @krshrimali

**Comment by krshrimali** [comment]
Thanks, @rajivnayanc for the contribution. LGTM. 🚀

**Comment by rajivnayanc** [comment]
Thank You! Looking forward to make more contributions. :)

**Pull request merged**
